### ISSUE TITLE
[Metrics] Partial AUROC

### ISSUE THREAD

**Issue by prampey** [PyTorchLightning/pytorch-lightning#3675]
## Feature

We currently have an `auroc` metric implementation.
Can we add an additional `max_fpr` parameter to enable a partial AUROC implementation?

## Motivation

Partial AUROC is often more useful in anomaly detection tasks to evaluate the appropriate portion of the ROC curve,
I stumbled onto Lightning metrics looking for a fast Tensor implementation of sklearn metrics. Since sklearn is heavily used, might be useful to reflect their API for metric implementations. 
Heres's the function signature of the sklearn version:
`sklearn.metrics.roc_auc_score(y_true, y_score, *, average='macro', sample_weight=None, max_fpr=None, multi_class='raise', labels=None)`

## Method
According to this code, I will need to filter through the return values from the roc method using max_fpr, before calculating area under the curve.
https://github.com/PyTorchLightning/pytorch-lightning/blob/d79bce1dffcc2d65799987608afb4a97d3090451/pytorch_lightning/metrics/functional/classification.py#L840-L874

**Comment by awaelchli** [comment]
@SkafteNicki @justusschock does it make sene to extend auroc this way?

**Comment by justusschock** [comment]
I think with a default of `None` this definitely makes sense. Want to give it a shot @prampey ?

**Comment by prampey** [comment]
Yes gladly.

**Comment by SkafteNicki** [comment]
Closing as this was implemented in #3790
